src/main/python/merchant_comparison.py

```
@@ -0,0 +1,1734 @@
+    """Enhanced ZIP code matching with 5-digit normalization"""
+    if pd.isna(zip1) or pd.isna(zip2):
+        return False
+    
+    # Convert to string and extract first 5 digits
+    z1 = str(zip1).strip()[:5]
+    z2 = str(zip2).strip()[:5]
+    
+    # Must be numeric and same
+    try:
+        return z1.isdigit() and z2.isdigit() and z1 == z2
+    except:
```

Ruff (0.12.2)

Do not use bare `except`

(E722)